bubble/kumar51/main.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+
+	end := len(s) - 1
+	for {
+		if end == 0 {
+			break
+		}
+
+		for i := 0; i < len(s)-1; i++ {
+			if s[i] > s[i+1] {
```

Comment:
Make a copy of `s` first, otherwise you will modify the input.
Write a test ensuring that the input does not get modified.